src/streamlink/plugins/adultswim.py

```diff
@@ -1,152 +1,135 @@
+import logging
 import re
 
 from streamlink.plugin import Plugin
-from streamlink.plugin.api import StreamMapper
 from streamlink.plugin.api import validate
-from streamlink.plugin.api import useragents
-from streamlink.stream import HDSStream
 from streamlink.stream import HLSStream
-from streamlink.stream import HTTPStream
-from streamlink.utils import parse_json, parse_xml
+from streamlink.utils import parse_json
+
+log = logging.getLogger(__name__)
 
 
 class AdultSwim(Plugin):
-    API_URL = "http://www.adultswim.com/videos/api/v2/videos/{id}?fields=stream"
-    vod_api = " http://www.adultswim.com/videos/api/v0/assets"
-
-    url_re = re.compile(r"""https?://(?:www\.)?adultswim\.com/videos
-            (?:/(streams))?
-            (?:/([^/]+))?
-            (?:/([^/]+))?
-            """, re.VERBOSE)
-    _stream_data_re = re.compile(r"(?:__)?AS_INITIAL_DATA(?:__)? = (\{.*?});", re.M | re.DOTALL)
-
-    live_schema = validate.Schema({
-        u"streams": {
-            validate.text: {u"stream": validate.text,
-                            validate.optional(u"isLive"): bool,
-                            u"archiveEpisodes": [{
-                                u"id": validate.text,
-                                u"slug": validate.text,
-                            }]}}
+    api_url = "https://api.adultswim.com/v1"
+    hash = "b27de189d91c06d5646dc7faea49282b97a4a25247e0927aa9bec7dd74ab6c71"
+    video_data_url = "https://www.adultswim.com/api/shows/v1/media/{0}/desktop"
+
+    json_data_re = re.compile(
+        '.*<script id="__NEXT_DATA__" type="application/json">'
+        '({.*?}})'
+        '</script>.*', re.S
```

Comment:
Noted regarding ```re.DOTALL``` and the change implemented.  I'm not a regex expert, but would also expect the ```.*``` at the beginning and end to be unnecessary, but the regex as it is does not match without them.  I think I'll have to revisit this to work it out a bit more...

Thanks.

Reply:
> but the regex as it is does not match without them

There's a difference between `re.match` and `re.search`:
- https://docs.python.org/3/library/re.html#re.Pattern.match
- https://docs.python.org/3/library/re.html#re.Pattern.search

By wrapping the entire regex in `.*`, you're implicitly turning every `match` call into a `search` call.

The greedy qualifier `*?` is also not needed and maybe even bad, because the site could add another script tag with JSON data and break your implementation with that. Unless the JSON data doesn't contain a `</script>` tag, the greedy qualifier is not needed.

Reply:
Ah!!!  Many thanks for that indeed.  It's now fixed.  I'm currently switching between fixes on this plugin and n13tv.  I will push changes on both when I have worked through all the issues you have identified.  Please bear with me.  As always, your feedback is much appreciated.  Thank you.